### ISSUE TITLE
`create` directive?

### ISSUE THREAD

**Issue by jesseleite** [anishathalye/dotbot#199]
I love the simplicity of how symlinking works, but is it possible to add a `create` directive, which would create a directory (and any parent directories, if necessary), _without symlinking_?  Right now I'm running this as a shell command:

```
  - description: Creating vim undo cache
    command: "[ -d ~/.vim/undo ] || mkdir ~/.vim/undo"
```

**Comment by anishathalye** [comment]
Sure, this seems reasonable. What should the format be?

```
- mkdir:
    - ~/.vim/undo
    - ~/path/to/some/other/folder
```

**Comment by jesseleite** [comment]
I would say:

```
- create:
    - ~/.vim/undo
    - ~/path/to/some/other/folder
    - ~/path/to/some/other/blankfile.txt
```

With `create` instead of `mkdir`, because you already abstract terminology like `shell` and `link`, opposed to `bash` and `ln`.  Also, I'm not sure when creating a blank file might be desirable, but why not support it as well?

**Comment by jesseleite** [comment]
Also not sure how far I'll get, but going to attempt a PR for Hacktoberfest 😅 ...if that's okay with you?

**Comment by anishathalye** [comment]
Sure, PRs are always appreciated 😄 

Agree re: abstract terminology.

Hmm, I don't know if we can handle both files and folders (also, are empty files useful?). Consider your example (repeated below for convenience):

```
- create:
    - ~/.vim/undo
    - ~/path/to/some/other/folder
    - ~/path/to/some/other/blankfile.txt
```

Here, how would Dotbot know that `folder` is supposed to be a folder, rather than a file with no file extension? Similarly, how would it know that `blankfile.txt` is a file, rather than a folder that happens to have a "." in the name?

**Comment by jesseleite** [comment]
I think files could be useful for creating blank log files, though I don't have a concrete example.  I figure if `link` can link both files and folders, why not `create` both files and folders?

That said, I think you're right about detecting file vs folder, and files should be an explicitly optional thing...

```
- create:
  - path: ~/.vim/undo
    recreate: true
  - path: ~/path/to/some/other/blank-log-file.txt
    recreate: true
    file: true
```

What are your thoughts on that?

**Comment by anishathalye** [comment]
What are the semantics of `recreate`?

**Comment by jesseleite** [comment]
Basically like `relink` and `force`, but for the new `create` directive.  Would remove the old target and re-create it.

**Comment by anishathalye** [comment]
I think `relink` and `force` make sense in the context of `link`, e.g. when there is a regular file, and you want it replaced with the symlink. I don't think `recreate` makes sense in the context of making a directory? The way I'd interpret `create: foo/bar` in a config is that the user wants that folder to be created. If it already exists, then do nothing. I can't think of cases where it makes sense to unlink it and then recreate it. (/ this seems to go against Dotbot's philosophy of idempotence)

**Comment by jesseleite** [comment]
Fair enough, I can leave the options out.  Even the `file: true` option could be left out until someone presents a valid use case for it.  I figured it could be useful, but don't have a concrete use case example.

**Comment by anishathalye** [comment]
Sounds good. Let’s leave out the file option until someone has a use case for it.

Let me know when your PR is ready for review :)

**Comment by jesseleite** [comment]
Closes by #200.
